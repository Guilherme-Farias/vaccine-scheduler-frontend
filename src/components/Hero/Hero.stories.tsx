import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Hero from '.';

export default {
  title: 'Hero',
  component: Hero,
} as ComponentMeta<typeof Hero>;

const Template: ComponentStory<typeof Hero> = args => <Hero {...args} />;

export const Default = Template.bind({});
Default.args = {};
