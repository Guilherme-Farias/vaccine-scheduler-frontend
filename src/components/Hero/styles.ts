import styled, { css } from 'styled-components/macro';

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    gap: ${theme.spacings.lg};
    height: 100%;

    @media ${theme.queries.tabletAndSmaller} {
      justify-content: start;
      flex-direction: column-reverse;
      align-items: center;
      gap: ${theme.spacings.sm};
    }
  `}
`;

export const Typography = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes['5xl']};
    font-weight: ${theme.font.regular};

    @media ${theme.queries.tabletAndSmaller} {
      font-size: ${theme.font.sizes['3xl']};
    }
  `}
`;

export const Highlight = styled.strong`
  ${({ theme }) => css`
    color: ${theme.colors.primary};
  `}
`;

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.sm};

    max-width: 300px;

    margin-top: ${theme.spacings['6xl']};
    @media ${theme.queries.tabletAndSmaller} {
      max-width: revert;
      align-items: revert;
      margin-top: ${theme.spacings['2xl']};
    }
  `}
`;
