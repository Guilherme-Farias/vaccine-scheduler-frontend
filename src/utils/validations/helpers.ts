import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
