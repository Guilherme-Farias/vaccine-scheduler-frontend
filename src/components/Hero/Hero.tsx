import React from 'react';
import { useNavigate } from 'react-router-dom';

import MaxWidthWrapper from 'components/MaxWidthWrapper';
import Button from 'components/Button';

import heroImage from 'assets/images/hero-img.svg';

import * as S from './styles';

const Hero: React.FC = () => {
  const navigate = useNavigate();
  return (
    <MaxWidthWrapper>
      <S.Content>
        <section>
          <S.Typography>
            Bem vindo ao <S.Highlight>Minha vacina</S.Highlight> o sistema de
            agendamento da vacina de COVID-19
          </S.Typography>

          <S.ButtonContainer>
            <Button onClick={() => navigate('/agendamentos')}>
              Liste os agendamentos
            </Button>
            <Button onClick={() => navigate('/agendamentos/novo')}>
              Faça seu agendamento
            </Button>
          </S.ButtonContainer>
        </section>

        <img
          src={heroImage}
          alt="Enfermeira vacinando um paciente"
          title="Ilustração de uma pessoa com máscara sendo vacinada, vetor criado por Storyset - www.freepik.com/people"
        />
      </S.Content>
    </MaxWidthWrapper>
  );
};
export default Hero;
