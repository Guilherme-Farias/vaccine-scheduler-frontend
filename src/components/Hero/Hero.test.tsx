import { render, screen, userEvent } from 'utils/tests/helpers';

import Hero from '.';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('<Hero />', () => {
  it('should render Hero ', () => {
    const { container } = render(<Hero />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should navigate to "/agendamentos"', async () => {
    render(<Hero />);

    await userEvent.click(
      screen.getByRole('button', { name: /liste os agendamentos/i }),
    );

    expect(mockNavigate).toHaveBeenCalledWith('/agendamentos');
  });

  it('should navigate to "/agendamentos/novo"', async () => {
    render(<Hero />);

    await userEvent.click(
      screen.getByRole('button', { name: /fa√ßa seu agendamento/i }),
    );

    expect(mockNavigate).toHaveBeenCalledWith('/agendamentos/novo');
  });
});
