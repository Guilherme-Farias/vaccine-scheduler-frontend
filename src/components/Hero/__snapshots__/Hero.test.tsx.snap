// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render Hero  1`] = `
.c0 {
  width: 100%;
  max-width: 81.25rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

.c5 {
  border: 2px solid transparent;
  border-radius: 0.625rem;
  font-weight: 500;
  font-size: 1.125rem;
  padding: 0.75rem 1.25rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c5:focus {
  outline-color: #D03E35;
  outline-offset: 4px;
}

.c6 {
  background-color: #D03E35;
  color: #FFFFFF;
}

.c6:hover {
  background-color: #bf352c;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.25rem;
  height: 100%;
}

.c2 {
  font-size: 2.5rem;
  font-weight: 400;
}

.c3 {
  color: #D03E35;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.75rem;
  max-width: 300px;
  margin-top: 3rem;
}

@media (max-width:68.75rem) {
  .c1 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 0.75rem;
  }
}

@media (max-width:68.75rem) {
  .c2 {
    font-size: 1.875rem;
  }
}

@media (max-width:68.75rem) {
  .c4 {
    max-width: revert;
    -webkit-align-items: revert;
    -webkit-box-align: revert;
    -ms-flex-align: revert;
    align-items: revert;
    margin-top: 1.75rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <section>
      <h2
        class="c2"
      >
        Bem vindo ao 
        <strong
          class="c3"
        >
          Minha vacina
        </strong>
         o sistema de agendamento da vacina de COVID-19
      </h2>
      <div
        class="c4"
      >
        <button
          class="c5 c6"
          color="primary"
        >
          Liste os agendamentos
        </button>
        <button
          class="c5 c6"
          color="primary"
        >
          Faça seu agendamento
        </button>
      </div>
    </section>
    <img
      alt="Enfermeira vacinando um paciente"
      src="hero-img.svg"
      title="Ilustração de uma pessoa com máscara sendo vacinada, vetor criado por Storyset - www.freepik.com/people"
    />
  </div>
</div>
`;
